cpp/src/arrow/record_batch.h

```diff
@@ -300,6 +311,11 @@ class ARROW_EXPORT RecordBatchReader {
   /// \brief finalize reader
   virtual Status Close() { return Status::OK(); }
 
+  /// \brief Get the device type for record batches this reader produces
+  ///
+  /// default implementation is to return ARROW_DEVICE_CPU
```

Comment:
Does `ARROW_DEVICE_CPU` even exist as a symbol?

Reply:
good point, it's a macro that exists only in `abi.h`, i'll fix the comment to reference the DeviceAllocationType